### ISSUE TITLE
Need meta params total_count, filter_count

### ISSUE THREAD

**Issue by uitlaber** [apstanisic/directus-dart#10]
I managed to add my custom fields. https://github.com/uitlaber/directus-dart/commit/10c97a97252415cc877a0ee703acf8bafd179906  Now I can search through collections or other methods. And I can add the meta parameter but the problem get response.  
[https://docs.directus.io/reference/api/query/meta.html](url)
I tried  edit directus_list_response.dart
```
// ignore: import_of_legacy_library_into_null_safe
import 'package:dio/dio.dart';
import 'package:directus/src/modules/items/items_converter.dart';
import 'package:directus/src/modules/items/map_items_converter.dart';

import 'data_classes.dart';

/// Directus response object when data is [List].
///
/// When response is returning a [List] of items, use this instead of [DirectusRespones].
/// This class is created to properly and type safely convert data.
/// Using [List] respones inside [DirectusRespone] will throw an error. Only this
/// response should be used. It has all the same properties and methods,
/// but adopted when response is [List].
class DirectusListResponse<T> implements DirectusResponse<List<T>> {
  /// Response data.
  @override
  late final List<T> data;

  List<T>? meta;

  /// Manually set data.
  DirectusListResponse.manually(data) : data = data;

  /// Constructor that converts Dio [Response] to [DirectusListResponse].
  /// You can pass converter that is used to convert response [Map] to
  /// proper object, by default it will return [Map].
  DirectusListResponse(Response dioResponse, {ItemsConverter? converter}) {
    converter ??= MapItemsConverter();
    var data = dioResponse.data['data'];
    var meta = dioResponse.data['meta'];

    print(data);

    if (!(data is List)) {
      throw DirectusError(message: 'Data should be a list.');
    } else {
      this.data =
          data.map((item) => converter!.fromJson(item)).cast<T>().toList();
    }

    if ((meta is List)) {
      this.meta =
          meta.map((item) => converter!.fromJson(item)).cast<T>().toList();
    }
  }

  /// Static method that you can pass a closure that should return Dio [Response].
  ///
  /// This will automaticaly convert [Response] to either [DirectusListResponse] or [DirectusError].
  /// Most of the methods that return JSON object with key `data` that is [List]
  /// should be called inside this method.
  /// You can pass converter that is used to convert response [Map] to
  /// proper object, by default it will return [Map].
  static Future<DirectusListResponse<U>> fromRequest<U>(
    Future<Response<dynamic>> Function() func, {
    ItemsConverter? converter,
  }) async {
    converter ??= MapItemsConverter();
    try {
      final response = await func();
      return DirectusListResponse<U>(response, converter: converter);
    } on DioError catch (error) {
      throw DirectusError.fromDio(error);
    }
  }
}

```

**Comment by apstanisic** [comment]
I've published 0.5.0 that has support for meta. 
Meta in `dioResponse` is `Map`, not `List`, that's why your code is not working. Is there a need for custom params in query?

You can now add meta to Query like the code bellow. It's a little more verbose then to simply pass `List<String>`, but provides better type safety.
```dart
final res = await sdk.readMany(query: Query(meta: Meta(totalCount: true, filterCount: true)));
print(res.meta.totalCount);
print(res.meta.filterCount);

```

**Comment by apstanisic** [comment]
Added in #11 and in a125419e1c7d41d6293542227380db6a3f501c0d
